wire-ios/Wire-iOS/Sources/UserInterface/DeviceView/Views/E2EIdentityCertificateDetailsView.swift

```diff
@@ -0,0 +1,152 @@
+//
+// Wire
+// Copyright (C) 2023 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import SwiftUI
+import WireCommonComponents
+
+struct E2EIdentityCertificateDetailsView: View {
+    @Environment(\.dismiss)
+    private var dismiss
+    private var borderColor: Color {
+        isDownloadAndCopyEnabled ?
+        SemanticColors.View.backgroundDefaultBlack.swiftUIColor : SemanticColors.View.backgroundDefaultWhite.swiftUIColor
+    }
+    var certificateDetails: String
+    var isDownloadAndCopyEnabled: Bool
+
+    @State var isMenuPresented: Bool
+
+    var performDownload: (
+        () -> Void
+    )?
+
+    var performCopy: (
+        (
+            String
+        ) -> Void
+    )?
+
+    private var titleView: some View {
+        HStack {
+            Spacer()
+            Text(L10n.Localizable.Device.Details.CertificateDetails.title)
+                .font(FontSpec.headerSemiboldFont.swiftUIFont)
+            Spacer()
+            SwiftUI.Button(
+                action: {
+                    dismiss()
+                },
+                label: {
+                    Image(.close)
+                        .foregroundColor(SemanticColors.Icon.foregroundDefaultBlack.swiftUIColor)
```

Comment:
Detecting light/dark mode using SwiftUI we need to use SwiftUI's .colorScheme. Can we use it here for now and then think of a more central solution. Please check: https://developermemos.com/posts/color-scheme-swiftui

Reply:
This works as expected now with Semantic Colors as I convert from `UIColor` to `Color`